### PR TITLE
#25504 healthcheck endpoint

### PR DESCRIPTION
**Pull request by dsilvam**
New healthcheck endpoint for Experiments. 

It returns false whether the App is bad configured or upon any other errors in the communication with Jitsu.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/experiments/ExperimentsResource.java [hunk 1]
```diff
@@ -500,6 +518,45 @@ public ResponseEntityExperimentResults getResult(@Context final HttpServletReque
...
+        }
+
+        final String url = analyticsApp.getAnalyticsProperties().analyticsWriteUrl();
+
+        final CircuitBreakerUrlBuilder builder = CircuitBreakerUrl.builder()
+                .setMethod(Method.POST)
+                .setUrl(url)
+                .setParams(map("token", analyticsApp.getAnalyticsProperties().analyticsKey()))
+                .setTimeout(4000)
+                .setHeaders(POSTING_HEADERS);
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/experiments/ExperimentsResource.java:549]: I think we should creare a Util for this, also we should check that it connect to CubeJs too.


Also we should check that this is not set analyticsApp.getAnalyticsProperties().analyticsKey() before hit the Jitsu or CubeJs End point

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/experiments/ExperimentsResource.java [hunk 2]
```diff
@@ -500,6 +518,45 @@ public ResponseEntityExperimentResults getResult(@Context final HttpServletReque
...
+
+        final Map<String, Object> testObject = Map.of("test", "test");
+
+        final EventsPayload eventPayload = new EventsPayload(testObject);
+        final Optional<Response> eventResponse = new EventLogRunnable(host, eventPayload)
+                .sendEvent(builder, new EventPayload(new JSONObject(testObject)));
+
+        return new ResponseEntityView<>(Map.of("healthy", eventResponse.isPresent()));
+    }
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/experiments/ExperimentsResource.java:558]: I don't know if we should judt send a TRUE|FALSE here, maybe we should send a JSON Object, something like:


{
     healthy: true|false,
     [any attribute name to say that is possible get the token]:  true|false,
    connection: {
       jitsu: true| false,
       cubejs: false
    }
}

**Review by freddyucv — CHANGES_REQUESTED** [review]
Postman test missing

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/experiments/ExperimentsResource.java (hunk 2 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/experiments/ExperimentsResource.java:558]: I think for now it is ok like this so the UI can tell if it should show the message or not. We can then enhance for v2

**Comment by dsilvam** [comment]
> Postman test missing

@freddyucv postman test is complicated since there's no analytics stack running to check true\false options. We could only test the "false" state. We can do that.

**Review by freddyucv — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

**Review by victoralfaro-dotcms — APPROVED** [review]

**Pull request merged**